src/pysdccc/_runner.py

```diff
@@ -0,0 +1,310 @@
+        command = self._prepare_command(
+            str(self.exe),
+            config=pathlib.Path(config),
+            requirements=pathlib.Path(requirements),
+            **kwargs,
+        )
+        try:
+            return_code = subprocess.run(command, timeout=timeout, check=True, cwd=self.exe.parent).returncode  # noqa: S603
+        except subprocess.CalledProcessError as e:
+            return_code = e.returncode
+        return (
+            return_code,
+            self._get_result(DIRECT_TEST_RESULT_FILE_NAME),
+            self._get_result(INVARIANT_TEST_RESULT_FILE_NAME),
+        )
+
+    def get_version(self) -> str:
+        """Get the version of the SDCcc executable."""
+        # use capture_output = True to get stdout and stderr instead of check_output which only collects stdout
+        process = subprocess.run([str(self.exe), '--version'], check=True, capture_output=True, cwd=self.exe.parent)  # noqa: S603
+        return process.stdout.decode(_common.ENCODING).strip()
+
+
+class SdcccRunnerAsync(_BaseRunner):
+    """Asynchronous runner for SDCcc."""
+
+    def __init__(self, test_run_dir: _common.PATH_TYPE, exe: _common.PATH_TYPE | None = None):
+        """Initialize the SdcccRunnerAsync object.
+
+        :param exe: The path to the SDCcc executable. Must be an absolute path.
+        :param test_run_dir: The path to the directory where the test run results are to be stored. Must be an absolute
+        path.
+        :raises ValueError: If the provided paths are not absolute.
+        """
+        super().__init__(test_run_dir=test_run_dir, exe=exe)
+
+    async def run(
+        self,
+        *,
+        config: _common.PATH_TYPE,
+        requirements: _common.PATH_TYPE,
+        **kwargs: _common.CMD_TYPE,
+    ) -> tuple[int, TestSuite | None, TestSuite | None]:
+        """Run the SDCcc executable using the specified configuration and requirements.
+
+        This method executes the SDCcc executable with the provided configuration and requirements files,
+        and additional command line arguments. It logs the stdout and stderr of the process and waits for the
+        process to complete or timeout.
+        Checkout more parameter under https://github.com/draegerwerk/sdccc?tab=readme-ov-file#running-sdccc
+
+        :param config: The path to the configuration file. Must be an absolute path.
+        :param requirements: The path to the requirements file. Must be an absolute path.
+        :param kwargs: Additional command line arguments to be passed to the SDCcc executable.
+        :return: A tuple containing the returncode of the SDCcc process, parsed direct and invariant test results as
+        TestSuite objects.
+        :raises ValueError: If the provided paths are not absolute.
+        :raises TimeoutError: If the process is running longer than the timeout.
+        """
+        command = self._prepare_command(config=pathlib.Path(config), requirements=pathlib.Path(requirements), **kwargs)
```

Comment:
1. If i get it right the path to the executable is missing

2. I have no log messages on my console when running the async example. Maybe i am doing sth. wrong. Other wise a solution may reading out stdout and log this?! On the other hand, the log messages are written to the SDCcc.log.

Reply:
fixed as discussed